File: Software Development Module/Revision/BMICalculator/BMICalculatorApp.java
/*
BMICalculatorApp.java
Billy Cussen
30/07/2020
*/

import java.util.Scanner;
import java.util.InputMismatchException;
import java.math.BigDecimal;

public class BMICalculatorApp{

    public static void main(String[]args){

        //Variables
        double height = 0.0;
        double weight = 0.0;
        String BMI;
        boolean inputCheck;

        //Object
        Scanner keyboard;
        BMICalculator myBMI;

        //Input
        do{
            inputCheck = true;
            keyboard = new Scanner(System.in);

            try{
                System.out.print("Please enter your height in Metres (e.g. 1.56): ");
                height = keyboard.nextDouble();

                if((BigDecimal.valueOf(height).scale() > 2)){
                    throw new InputMismatchException();
                }

            }catch(InputMismatchException | NumberFormatException e){
                inputCheck = false;
                System.out.println("Please enter your Height in Metres with up to 2 Decimal Places please (e.g. 1.53)");
            } finally{
                keyboard = null;
            }

        }while(!inputCheck);

        do{
            inputCheck = true;
            keyboard = new Scanner(System.in);

            try{
                System.out.print("Please enter your Weight in KG (e.g. 70.5): ");
                weight = keyboard.nextDouble();

                if((BigDecimal.valueOf(weight).scale() > 2)){
                    throw new InputMismatchException();
                }


            }catch(InputMismatchException | NumberFormatException e){
                inputCheck = false;
                System.out.println("Please enter your Weight in KG with up to 2 Decimal Places please (e.g. 70.50)");
            }

        }while(!inputCheck);

        myBMI = new BMICalculator(weight, height);
        BMI = myBMI.getBMI();
        myBMI = null;
        System.out.println(BMI);
    }

}

File: Python/Software Development Module - Python/Revision/BMICalculator/BMICalculatorApp.py
"""
BMICalculatorApp.py
Billy Cussen
21/10/2020
"""

#Objects
import gc
import BMICalculator
from BMICalculator import myBMICalculator

#Variables
repeat = "y"

while repeat == "y":

    inputCheck = False

    #Input - Weight?
    while inputCheck == False:
        try:
            weight = float(input("Please enter Weight in KG: "))
            if weight <= 0:
                raise Exception
        except Exception as e:
            print("Please enter a Valid Number as your weight (e.g. 82)")
        else:
            inputCheck = True

    #Input - Height?
    while inputCheck:
        try:
            height = float(input("Please enter your your Height in Metres (e.g. 1.7): "))
            if height <= 0:
                raise Exception
        except Exception as e:
            print("You need to enter a Valid Number as your Height (e.g. 1.8)!")
        else:
            inputCheck = False

    #Process and Output
    myBMICalculatorApp = myBMICalculator(weight,height)
    print(myBMICalculatorApp.calculateBMI())
    myAirlineApp = None
    gc.collect()

    repeat = ""
    while repeat.lower() != "y" and repeat.lower() != "n":
        repeat = str(input("Please enter \"y\" to repeat and \"n\" to exit: "))
        if repeat.lower() != "y" and repeat.lower != "n":
            print("You need to enter \"y\" or \"n\"")

print("Thanks for using my BMI Calculator App")